src/vcpkg/configuration.cpp

```diff
@@ -387,6 +382,81 @@ namespace
         return ret;
     }
 
+    std::vector<LocalizedString> collect_package_pattern_warnings(const std::vector<RegistryConfig>& registries)
+    {
+        struct LocationAndRegistry
+        {
+            StringView location;
+            StringView registry;
+        };
+
+        // handle warnings from package pattern declarations
+        std::map<StringView, std::vector<LocationAndRegistry>> patterns;
```

Comment:
Key of `StringView` is a bit scary in terms of ownership? Might be worth either using `string` instead or commenting on how the lifetime works.